test/e2e_node/pod_resize_test.go

```diff
@@ -307,56 +309,161 @@ func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework
 		}
 		return nil
 	}
+
+	var podCPURequestMilliValue, podCPULimitMilliValue, podMemoryLimitInBytes int64
 	for _, ci := range tcInfo {
-		if ci.Resources == nil {
-			continue
-		}
-		tc, _ := makeTestContainer(ci)
-		if tc.Resources.Limits != nil || tc.Resources.Requests != nil {
-			var expectedCPUShares int64
-			var expectedCPULimitString, expectedMemLimitString string
-			expectedMemLimitInBytes := tc.Resources.Limits.Memory().Value()
-			cpuRequest := tc.Resources.Requests.Cpu()
-			cpuLimit := tc.Resources.Limits.Cpu()
-			if cpuRequest.IsZero() && !cpuLimit.IsZero() {
-				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
-			} else {
-				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
-			}
-			cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
-			if cpuLimit.IsZero() {
-				cpuQuota = -1
-			}
-			expectedCPULimitString = strconv.FormatInt(cpuQuota, 10)
-			expectedMemLimitString = strconv.FormatInt(expectedMemLimitInBytes, 10)
-			if podOnCgroupv2Node {
-				if expectedCPULimitString == "-1" {
-					expectedCPULimitString = "max"
+		var cpuRequest, cpuLimit *resource.Quantity
+		var expectedMemLimitInBytes int64
+		if ci.Resources != nil {
```

Comment:
nit (to reduce nested `if`s):
```suggestion
		if ci.Resources == nil {
			continue
		}

		// rest of the logic
```

Reply:
I reduced nested `if`s.